oas

```diff
@@ -490,11 +533,93 @@ components:
         archived: false
         name: 'Marta'
         public_identifier: 'DE 123456789'
-        farm_identifier: '674'
+        alternative_identifier: '674'
         pen_urn: 'urn:animal_service:FDBE6F55-6179-4B88-9177-4AFBED33ADAC'
-        date_birth: '2021-03-06T12:34:45Z'
         sex: 'f'
         species: 'cattle'
+
+    Event:
+      type: object
+      properties:
+        date:
+          type: string
+          format: date-time
+          description: date when this event happened
+        category:
+          $ref: '#/components/schemas/EventCategory'
+        animal_urn:
+          $ref: '#/components/schemas/URN'
+        payload: 
+          oneOf:
+            - description: birth event
+              type: object
+              properties:
+                mother_public_identifier:
+                  type: string
+                father_public_identifier:
+                  type: string
+            - description: arrival event
+              type: object
+              properties:
+                 arrival_reason:
+                  type: string
+                  description: reason why this animal arrives on the farm, **todo** extend enum
+                  enum:
+                    - 'buyed'
+            - description: departure event
+              type: object
+              properties:
+                 departure_reason:
+                  type: string
+                  description: reason why this animal arrives on the farm, **todo** extend enum
+                  enum:
+                    - 'death'
+                    - 'sold'
+                    - 'slaughered'
+            - description: milk-report event
+              type: object
+              properties:
+                milk:
+                  type: number
+                  description: amount of milk given in kg
+                milkFat:
+                  type: number
+                  description: in %
+                milkProtein:
+                  type: number
+                  description: in %
+    AnimalWeight:
+      type: object
+      required:
+        - animal_urn
+        - average_weight
+        - median_weight
+        - date
+      properties:
+        animal_urn:
+          type: string
+          example: urn:animal_service:FDBE6F55-6179-4B88-9177-4AFBED33ADAC
+          description: Uniform resource name of an object with the syntax urn:<service_id>:<internal_id>. <service_id> must be the urn of a valid and active ATLAS service.
+        average_weight:
+          description: average weight of the animal in kg
+          type: number
+        calculated_weight:
+          description: calculated weight of the animal in kg
+          type: number
+        daily_weight_gain:
```

Comment:
What does a daily gain of weight mean in the context of a particular date? Is it the average of kg gained per day up to that date, or the average of kg gained per day up until today?

In any case, it seems clients could determine this value on their own by dividing the difference of weight on different dates by the number of days.

Reply:
true, I have to check that with the partner. if its just an easy comparison between current and last entry we could remove it here.

Reply:
its the weight gain calculated over the last 14days. indeed that could also be done by the clients but requesting all weighing events from last 14days causes some overhead. as it is an optional field and has value for our further project-goals I would keep it here.